x/interchainqueries/keeper/grpc_query.go

```diff
@@ -54,11 +54,18 @@ func (k Keeper) GetRegisteredQueries(ctx sdk.Context, req *types.QueryRegistered
 			passedConnectionIDFilter = req.GetConnectionId() == "" || query.ConnectionId == req.GetConnectionId()
 		)
 
-		if passedOwnerFilter && passedConnectionIDFilter {
+		// if result does not satisfy the filter, return (false, nil) to tell FilteredPaginate method to skip this value
+		if !(passedOwnerFilter && passedConnectionIDFilter) {
+			return false, nil
+		}
+
+		// when accumulate equals true, it means we are in the right offset/limit position
+		// so we check value satisfies the filter and add it to the final result slice
+		if accumulate && passedOwnerFilter && passedConnectionIDFilter {
```

Comment:
seems that its second and unnecessary check for `passedOwnerFilter && passedConnectionIDFilter`

Reply:
also maybe just do early return here? (just suggestion)
```
if !accumulate {
  return true, nil
}

queries = append(queries, query)
...
```

Reply:
> seems that its second and unnecessary check for passedOwnerFilter && passedConnectionIDFilter

Yes, but I think this improves readability, because it explicitly describe the condition of final result accumulation.

> also maybe just do early return here?

The same about readability. I think early return will decrease readability here, because current solution explicitly says when accumulate the result.